Terminal.Gui/Views/ColorPicker.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System;
+using NStack;
+
+namespace Terminal.Gui {
+
+	/// <summary>
+	/// The <see cref="ColorPicker"/> <see cref="View"/> Color picker.
+	/// </summary>
+	public class ColorPicker : View {
+		/// <summary>
+		/// Number of colors on a line.
+		/// </summary>
+		private static readonly int colorsPerLine = 8;
+
+		/// <summary>
+		/// Number of color lines.
+		/// </summary>
+		private static readonly int lineCount = 2;
+
+		/// <summary>
+		/// Horizontal zoom.
+		/// </summary>
+		private static readonly int horizontalZoom = 4;
+
+		/// <summary>
+		/// Vertical zoom.
+		/// </summary>
+		private static readonly int verticalZoom = 2;
+
+		// Cursor runes.
+		private static readonly Rune [] cursorRunes = new Rune []
+		{
+			0x250C, 0x2500, 0x2500, 0x2510,
+			0x2514, 0x2500, 0x2500, 0x2518
+		};
+
+		/// <summary>
+		/// Cursor for the selected color.
+		/// </summary>
+		public Point Cursor {
+			get {
+				return new Point (selectColorIndex % colorsPerLine, selectColorIndex / colorsPerLine);
+			}
+
+			set {
+				var colorIndex = value.Y * colorsPerLine + value.X;
+				SelectedColor = (Color)colorIndex;
+			}
+		}
+
+		/// <summary>
+		/// Fired when a color is picked.
+		/// </summary>
+		public event Action ColorChanged;
+
+		private int selectColorIndex = (int)Color.Black;
+
+		/// <summary>
+		/// Selected color.
+		/// </summary>
+		public Color SelectedColor {
+			get {
+				return (Color)selectColorIndex;
+			}
+
+			set {
+				selectColorIndex = (int)value;
+				ColorChanged?.Invoke ();
+				SetNeedsDisplay ();
+			}
+		}
+
+		/// <summary>
+		/// Initializes a new instance of <see cref="ColorPicker"/>.
+		/// </summary>
+		public ColorPicker () : this (0, 0, "Color Picker")
```

Comment:
Please add a constructor that takes just a Title (you'll see why as I add more comments).

Reply:
Yes, something was wrong. Thanks. I added this constructor.